File: LeetCode/stack/p921/c++/solution.cpp
class Solution {
public:
int change(char c){
    if(c==')')return '(';
    return ' ';
}
    int minAddToMakeValid(string s) {
        stack<char> k;
        for(char c:s){
            if(k.empty()){
                k.push(c);
                continue;
            }
            if(change(c)==k.top()){
                k.pop();
            }else{
                k.push(c);
            }
        }
        return k.size();
    }
};

File: LeetCode/stack/p921/java/solution.java
class Solution {
    public int minAddToMakeValid(String s) {
        int left = 0,right = 0;
        for(int i=0;i<s.length();i++) {
            if(s.charAt(i)=='('){
                left++;
            }else if(s.charAt(i)==')'){
                left--;
            }
            if(left==-1){
                right++;
                left++;
            }
        }
        return left+right;
    }
}